sdk/core/azure-core/src/main/java/com/azure/core/http/policy/OverrideHeaderPolicy.java

```diff
@@ -0,0 +1,33 @@
+package com.azure.core.http.policy;
+
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.Context;
+import com.azure.core.http.HttpRequest;
+import static com.azure.core.util.Context.AZURE_REQUEST_HTTP_HEADERS_KEY;
+import reactor.core.publisher.Mono;
+
+import java.util.Objects;
+
+/**
+ * The pipeline policy that override or add  {@link HttpHeader} in {@link HttpRequest} by reading values from
+ * {@link Context} with key  'azure-request-override-http-headers-key'.
+ */
+public class OverrideHeaderPolicy implements HttpPipelinePolicy {
```

Comment:
This is to satisfy the ask by App Config to be able to provide values for following http headers
• "x-ms-client-request-id"
• "x-ms-correlation-request-id"
• "correlation-context"

End user need to add `HttpHeaders`  in Context object with a specific key to achieve this.

Reply:
Can we have the policy to name as `AddHeadersFromContextPolicy` or something that indicates that headers are coming from the Context?